AK/Types.h

```diff
@@ -19,6 +19,28 @@ using i32 = __INT32_TYPE__;
 using i16 = __INT16_TYPE__;
 using i8 = __INT8_TYPE__;
 
+#ifndef KERNEL
+using f32 = float;
+static_assert(__FLT_MANT_DIG__ == 24 && __FLT_MAX_EXP__ == 128);
+
+using f64 = double;
+static_assert(__DBL_MANT_DIG__ == 53 && __DBL_MAX_EXP__ == 1024);
+
+// FIXME: clang does not declare __FLT128_* but provides __float128.
+#    if defined(__FLT128_MANT_DIG__) && defined(__FLT128_MAX_EXP__) && __FLT128_MANT_DIG__ == 113 && __FLT128_MAX_EXP__ == 16384
+#        define AK_HAS_FLOAT_128 1
+using f128 = __float128;
+#    endif
+
+#    if __LDBL_MANT_DIG__ == 64 && __LDBL_MAX_EXP__ == 16384
```

Comment:
Can't we just use `__LDBL_MANT_DIG__` in `FloatExtractor` itself and skip this pile of macros?

Reply:
The problem is it is not possible to fit mantissa in u64 if `__LDBL_MANT_DIG__ == 113`. Moreover, float-to-string conversion really depends on the exact size, so it's nice to implement it, for example, for f80 and not for some arbitrary `long double`.